controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -590,21 +244,9 @@ func (r *StorageConsumerReconciler) verifyCephResourcesDoNotExist() bool {
 			cephClient := &rookCephv1.CephClient{}
 			cephClient.Name = cephResource.Name
 			cephClient.Namespace = r.namespace
-			if err := r.get(cephClient); err == nil || !errors.IsNotFound(err) {
-				return false
-			}
-		case "CephBlockPool":
-			cephBlockPool := &rookCephv1.CephBlockPool{}
-			cephBlockPool.Name = cephResource.Name
-			cephBlockPool.Namespace = r.namespace
-			if err := r.get(cephBlockPool); err == nil || !errors.IsNotFound(err) {
-				return false
-			}
-		case "CephFilesystemSubVolumeGroup":
-			cephFilesystemSubVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
-			cephFilesystemSubVolumeGroup.Name = cephResource.Name
-			cephFilesystemSubVolumeGroup.Namespace = r.namespace
-			if err := r.get(cephFilesystemSubVolumeGroup); err == nil || !errors.IsNotFound(err) {
+			err := r.get(cephClient)
+			cephUserType := cephClient.Annotations[StorageCephUserTypeAnnotation]
+			if cephUserType == "healthchecker" && (err == nil || !errors.IsNotFound(err)) {
```

Comment:
This only takes care of the cephClient, but what about pending claims? where do you check there are no claims for the consumer? 

Also, I was unable to find the code that adds the claims to the resource list?

Reply:
can I do it in deletion PR since this part will be triggered during deletion and this change was part of refactoring? I'll put this comment on deletion pr if we agree

Reply:
Ok